### PR TITLE
feat(frontend): settings screen bb-334

### PR DESCRIPTION
**Pull request by ihorLysak**
### Summary
created settings screen
Screenshot 2024-09-10 at 16 27 39

### CODE REVIEW

File: apps/frontend/src/index.tsx [hunk 1]
```diff
@@ -78,6 +63,31 @@ createRoot(document.querySelector("#root") as HTMLElement).render(
...
+								element: (
+									<ProtectedRoute
+										component={<Settings />}
+										redirectTo={AppRoute.SIGN_IN}
+									/>
+								),
+								path: AppRoute.SETTINGS,
+							},
+							{
+								element: <Auth />,
```
Comment (ihorLysak) [inline apps/frontend/src/index.tsx:76]: moved them down because they are not protected

File: apps/frontend/src/assets/img/sign-out.svg
Comment (fshabanov) [inline apps/frontend/src/assets/img/sign-out.svg:1]: Why was it removed?

File: apps/frontend/src/modules/users/slices/actions.ts [hunk 2]
```diff
@@ -26,11 +27,17 @@ const saveNotificationAnswers = createAsyncThunk<
 	UserDto,
 	NotificationAnswersPayloadDto,
 	AsyncThunkConfig
->(`${sliceName}/saveNotificationAnswers`, async (payload, { extra }) => {
-	const { usersApi } = extra;
+>(
+	`${sliceName}/saveNotificationAnswers`,
```
Comment (fshabanov) [inline apps/frontend/src/modules/users/slices/actions.ts:31]:
```suggestion
	`${sliceName}/save-notification-answers`,
```

File: apps/frontend/src/pages/libs/constants/constants.ts [hunk 3]
```diff
@@ -0,0 +1,4 @@
+export {
+	NOTIFICATION_FREQUENCY_OPTIONS,
+	TASK_DAYS_OPTIONS,
+} from "./notification-questions-options.constant.js";
```
Comment (fshabanov) [inline apps/frontend/src/pages/libs/constants/constants.ts:4]: Create each at a separate file

File: apps/frontend/src/pages/libs/constants/constants.ts [hunk 4]
```diff
@@ -0,0 +1,4 @@
+export {
```
Comment (fshabanov) [inline apps/frontend/src/pages/libs/constants/constants.ts:1]: We shouldn't have ~/pages/libs. If it is something global, move to ~/libs

**Review by fshabanov — CHANGES_REQUESTED** [review]
Let's have week days in 2 rows and frequency answers shorter, they are too long

File: apps/frontend/src/assets/img/sign-out.svg
Comment (ihorLysak) [reply apps/frontend/src/assets/img/sign-out.svg:1]: We don't use it anywhere

**Comment by ihorLysak** [comment]
Screenshot 2024-09-11 at 11 11 05

File: apps/frontend/src/libs/types/types.ts [hunk 5]
```diff
@@ -9,6 +9,7 @@ export {
 } from "./form-field-properties.type.js";
 export { type IconName } from "./icon-name.type.js";
 export { type InputOption } from "./input-option.type.js";
+export { type NotificationQuestionsFormValues } from "./notification-questions-form-values.types.js";
```
Comment (fshabanov) [inline apps/frontend/src/libs/types/types.ts:12]:
```suggestion
export { type NotificationQuestionsFormValues } from "./notification-questions-form-values.type.js";
```

File: apps/frontend/src/pages/settings/libs/constants/settings-notification-frequency-options.constant.ts [hunk 6]
```diff
@@ -0,0 +1,14 @@
+import { NotificationFrequency } from "~/modules/users/users.js";
+
+const SETTINGS_NOTIFICATION_FREQUENCY_OPTIONS = [
```
Comment (fshabanov) [inline apps/frontend/src/pages/settings/libs/constants/settings-notification-frequency-options.constant.ts:3]: We already have `NOTIFICATION_FREQUENCY_OPTIONS`, which has the same values

**Review by fshabanov — CHANGES_REQUESTED** [review]
Center everything in the layout

**Comment by ihorLysak** [comment]
Screenshot 2024-09-11 at 14 20 04
Centered layout, extendded notification options width for symmetry

File: apps/frontend/src/modules/auth/slices/actions.ts [hunk 7]
```diff
@@ -73,7 +73,9 @@ const requestResetPassword = createAsyncThunk<
 	EmailDto,
 	AsyncThunkConfig
 >(`${sliceName}/send-reset-password-link`, async (emailPayload, { extra }) => {
-	const { authApi, notification } = extra;
+	const { authApi, notification, storage } = extra;
+
+	await storage.drop(StorageKey.TOKEN);
```
Comment (fshabanov) [inline apps/frontend/src/modules/auth/slices/actions.ts:78]: I don't think it should be here. Also, not part of your task

**Review by fshabanov — CHANGES_REQUESTED** [review]

**Pull request merged**